ui/app/js/controllers/release_new_controller.js

```diff
@@ -18,6 +18,28 @@ function($scope, $http, $modalInstance, CSRF, Releases, releases) {
   $scope.errors = {};
   $scope.saving = false;
 
+  $scope.fillName = function () {
+    var file = $scope.dataFile;
+
+    var reader = new FileReader();
+    reader.onloadend = function(evt) {
+      var blob = evt.target.result;
+      $scope.$apply( function() {
+         $scope.release.name = JSON.parse(blob).name;
+      });
+    };
+    if (typeof file !== 'undefined') {
+      // should work
+      reader.readAsText(file);
+    }
+
+  };
+
+  $scope.changeName = function () {
+    //wait for actual file to be loaded
+    setTimeout( $scope.fillName , 500);
```

Comment:
I think this might go away if you don't go through angular.element to call it...see if that helps. We do this sort of thing in lots of places, eg: when addRelease is called on releases.html